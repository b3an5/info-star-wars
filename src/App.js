import React, { Component } from 'react';
import Splash from './components/splash/splash.js';
import Header from './components/header/header.js';
import Content from './components/content/content.js';
import { getFilms, getPeople, getVehicles, getPlanets } from './apiCalls.js'

class App extends Component {
  constructor() {
    super();
    this.state = {
      isLoading: true,
      splash: true,
      film: null,
      randomFilm: null,
      data: {
        People: [],
        Planets: [],
        Vehicles: []
      },
      cardCategory: 'People'
    }
  }

  switchCardCat = (e) => {
    this.setState({cardCategory: e.target.innerText})
  }

  componentDidMount() {
    getFilms()
      .then(data => {
        this.setState({
          film: data.results,
          isLoading: false,
          randomFilm: Math.floor(Math.random() * data.results.length)
        });
      })
      .catch(error => console.log(error));

    getPeople()
      .then(results => this.setState( ({ data }) => {
        data.People = results
        return data
      } ))
      .catch(error => console.log(error))

  getVehicles()
    .then(results => this.setState( ({ data }) => {
      data.Vehicles = results.results
      return data
    }))
    .catch(error => console.log(error))

  getPlanets()
    .then(results => this.setState(({ data }) => {
      data.Planets = results
      return data
    }))
    .catch(error => console.log(error))
  }

  showMainPage = () => {
    this.setState({
      splash: false
    })
  }

  render() {
    if(this.state.isLoading) {
      return (
        <div className="App">
          <h1>...loading</h1>
        </div>
      );
    } 
    if(this.state.splash) {
      return (
        <div>
          <Header 
            splash={this.state.splash}
          />
          <Splash 
            film={this.state.film[this.state.randomFilm]}
            splashButton={this.showMainPage}
          />
        </div>
      )
    } else {
      return (
        <div>
          <Header 
            splash={this.state.splash}
            switchCardCat={this.switchCardCat}
          />
          <Content 
            cardCategories={this.state.data[this.state.cardCategory]}
            categoryName={this.state.cardCategory}
          />
        </div>
      )
    }
  }
}

export default App;
