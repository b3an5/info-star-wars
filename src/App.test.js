import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'
import { mockData } from './mockData.js'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('app', () => {
  let wrapper 
  let e
  const { films, people, planets, vehicles } = mockData;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    expect(wrapper.state()).toEqual({
      isLoading: true,
      splash: true,
      film: null,
      randomFilm: null,
      data: {
        People: [],
        Planets: [],
        Vehicles: []
      },
      cardCategory: 'People'
    })
  })

  it('should set cardCategory in state to innertext when switch cardCat is called', () => {
    e = {
      target: {
        innerText: 'Vehicles'
      }
    }
    expect(wrapper.state()).toEqual({
      isLoading: true,
      splash: true,
      film: null,
      randomFilm: null,
      data: {
        People: [],
        Planets: [],
        Vehicles: []
      },
      cardCategory: 'People'
    })
    wrapper.instance().switchCardCat(e)
    expect(wrapper.state()).toEqual({
      isLoading: true,
      splash: true,
      film: null,
      randomFilm: null,
      data: {
        People: [],
        Planets: [],
        Vehicles: []
      },
      cardCategory: 'Vehicles'
    })
  })

  it.skip('should populate state when component did mount runs', () => {
    wrapper.instance().componentDidMount()
    expect(wrapper.state()).toEqual({
        isLoading: true,
        splash: true,
        film: null,
        randomFilm: null,
        data: {
          People: people.results,
          Planets: planets.results,
          Vehicles: vehicles.results
        },
        cardCategory: 'People'
    })
  })

  it('should switch state splash to false when showMainPage is called', () => {
    wrapper.instance().showMainPage()
    expect(wrapper.state()).toEqual({
        isLoading: true,
        splash: false,
        film: null,
        randomFilm: null,
        data: {
          People: [],
          Planets: [],
          Vehicles: []
        },
        cardCategory: 'People'
    })
  })

  it('should match snapshot when splash is false', () => {
    wrapper.instance().showMainPage()
    expect(wrapper).toMatchSnapshot()
  })

})