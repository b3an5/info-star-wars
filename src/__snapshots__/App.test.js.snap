// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should match snapshot 1`] = `ShallowWrapper {}`;

exports[`app should match snapshot when splash is false 1`] = `ShallowWrapper {}`;
